### PR TITLE
[RDKDEV-262] DeviceIdentification: syncup from metro version

### PR DESCRIPTION
**Pull request by HaseenaSainul**
1. Module name corrected
2. Generic implementation added for Linux and Broadcom
3. RPI: device id fetching support added
4. Interface aligned for Amlogic, RealTek, RPI and MFR
5. Cleanup of Deactivation sequence
6. cmake: cleanup and depedency added from cmake inside DeviceIdentification/cmake

### CODE REVIEW

**Comment by mhughesacn** [comment]
Hi Haseena: This repo is assigned to RDK and needs RDK Apache headers, not Metrological - Martin

**Comment by HaseenaSainul** [comment]
> Hi Haseena: This repo is assigned to RDK and needs RDK Apache headers, not Metrological - Martin

Copyright updated

**Comment by mhughesacn** [comment]
Many thanks Haseena!

**Comment by npoltorapavlo** [comment]
I believe this breaks a unit test

**Comment by HaseenaSainul** [comment]
These changes will work only with the thunder changes add with the hash 331ee30b1eaf82a7e8944ee00b8cb1d1d0aca682. So @npoltorapavlo, could you please update unit test case to use the Thunder corresponding to this hash from R2 branch ?

**Comment by npoltorapavlo** [comment]
As your change breaks the tests, can you fix the tests. Thanks

**Comment by HaseenaSainul** [comment]
> As your change breaks the tests, can you fix the tests. Thanks

@npoltorapavlo Could you please help me to find the location of test ?
From the details it showing error:

**Comment by HaseenaSainul** [comment]
/home/runner/work/rdkservices/rdkservices/rdkservices/DeviceIdentification/DeviceIdentification.cpp:146:51: error: ‘struct WPEFramework::PluginHost::ISubSystem::IIdentifier’ has no member named ‘FirmwareVersion’
[40](https://github.com/rdkcentral/rdkservices/runs/5987986247?check_suite_focus=true#step:20:40)
  146 |         deviceInfo.Firmwareversion = _identifier->FirmwareVersion();
[41](https://github.com/rdkcentral/rdkservices/runs/5987986247?check_suite_focus=true#step:20:41)
      |                                                   ^~~~~~~~~~~~~~~
[42](https://github.com/rdkcentral/rdkservices/runs/5987986247?check_suite_focus=true#step:20:42)
/home/runner/work/rdkservices/rdkservices/rdkservices/DeviceIdentification/DeviceIdentification.cpp:147:[43](https://github.com/rdkcentral/rdkservices/runs/5987986247?check_suite_focus=true#step:20:43): error: ‘struct WPEFramework::PluginHost::ISubSystem::IIdentifier’ has no member named ‘Chipset’
43
  147 |         deviceInfo.Chipset = _identifier->Chipset();
[44](https://github.com/rdkcentral/rdkservices/runs/5987986247?check_suite_focus=true#step:20:44)
      |                                           ^~~~~~~
[45](https://github.com/rdkcentral/rdkservices/runs/5987986247?check_suite_focus=true#step:20:45)
make[2]: *** [DeviceIdentification/CMakeFiles/WPEFrameworkDeviceIdentification.dir/build.make:66: DeviceIdentification/CMakeFiles/WPEFrameworkDeviceIdentification.dir/DeviceIdentification.cpp.o] Error 1
[46](https://github.com/rdkcentral/rdkservices/runs/5987986247?check_suite_focus=true#step:20:46)
make[1]: *** [CMakeFiles/Makefile2:435: DeviceIdentification/CMakeFiles/WPEFrameworkDeviceIdentification.dir/all] Error 2
[47](https://github.com/rdkcentral/rdkservices/runs/5987986247?check_suite_focus=true#step:20:47)
make: *** [Makefile:133: all] Error 2
But that is related to the usage of wrong thunder version

**Review by MFransen69 — APPROVED** [review]

**Comment by anand-ky** [comment]
@HaseenaSainul @npoltorapavlo - Let's get the thunder R2 version merged to sprint first and then we can update unit tests with the new version. Will work with Karuna to get R2 updated.

**Comment by npoltorapavlo** [comment]
It builds now but fails in runtime. DeviceIdentification::Initialize calls IShell::Register which needs COMLink() to be valid. I.e. need to mock COMLink().

**Comment by HaseenaSainul** [comment]
@npoltorapavlo, @anand-ky, please wait to get @MFransen69 approval before doing the merge.

File: test/tests/DeviceIdentificationTest.cpp
Comment (MFransen69) [inline test/tests/DeviceIdentificationTest.cpp:67]: Now it is not my unittest but wouldn't it be better to just make one test out of this or if it is really desired to split it between a registered and a property test can me make helper funtions to reduce the duplicate code that now can be found in both tests to setup and tear down the mock code and plugin before the actual test is done? That will be much better for maintainability of the test code

**Review by MFransen69 — CHANGES_REQUESTED** [review]

**Review by MFransen69 — APPROVED** [review]

**Review by MFransen69 — APPROVED** [review]

**Comment by Firdose2109** [comment]
Can one of the admins verify this patch?

**Review by anand-ky — APPROVED** [review]

**Pull request merged**
